src/lib/bootloader/grub2base.rb

```diff
@@ -227,6 +216,20 @@ def serial_console?
 
   private
 
+    def propose_grub_default
+      if grub_default.kernel_params.empty?
+        kernel_line = Yast::BootArch.DefaultKernelParams(propose_resume)
+        grub_default.kernel_params.replace(kernel_line)
+      end
+      grub_default.gfxmode ||= "auto"
+      grub_default.recovery_entry.disable unless grub_default.recovery_entry.defined?
+      grub_default.distributor ||= ""
+      grub_default.default = "saved"
+      # always propose true as grub2 itself detect if btrfs used
+      grub_default.generic_set("SUSE_BTRFS_SNAPSHOT_BOOTING", "true")
+      grub_default.hidden_timeout = "0"
```

Comment:
as timeout and hidden timeout are quite related I would add this code to propose_timeout method https://github.com/yast/yast-bootloader/pull/672/files#diff-5797ed1429ad469f007496587dfc38e58cb54f82bc44c3bf99d81b9bef6b58beR366